integration-tests/dataproc_test_case.py

```diff
@@ -0,0 +1,137 @@
+        if scopes:
+            args.append("--scopes {}".format(scopes))
+        if metadata:
+            args.append("--metadata={}".format(metadata))
+        if dataproc_version:
+            args.append("--image-version={}".format(dataproc_version))
+        if timeout_in_minutes:
+            args.append("--initialization-action-timeout {}m".format(timeout_in_minutes))
+        args.append("--initialization-actions {}".format(init_action))
+        cmd = "gcloud dataproc clusters create {}".format(self.name)
+        for flag in args:
+            cmd += " {}".format(flag)
+        cmd += " --format=json"
+
+        ret_val, stdout, stderr = self.run_command(cmd, timeout_in_minutes=timeout_in_minutes or DEFAULT_TIMEOUT)
+
+        self.assertEqual(ret_val, 0, "Failed to create Cluster {}. Error: {}".format(
+            self.name,
+            stderr
+        ))
+        self.cluster_version = json.loads(stdout).get("config", {}).get("softwareConfig", {}).get("imageVersion")
+
+    def tearDown(self):
+        cmd = "yes | gcloud dataproc clusters delete {} --quiet".format(self.name)
+        ret_val, stdout, stderr = self.run_command(cmd)
+        self.assertEqual(ret_val, 0, "Failed to delete cluster {}. Error: {}".format(
+            self.name,
+            stderr
+        ))
+
+    def getClusterName(self):
+        return self.name
+
+    def upload_test_file(self, name):
+        cmd = 'gcloud compute scp {} {}:'.format(
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)),
+                self.COMPONENT.lower(),
+                self.TEST_SCRIPT_FILE_NAME
+            ),
+            name,
+        )
+        ret_code, stdout, stderr = self.run_command(cmd)
+        self.assertEqual(ret_code, 0, "Failed to upload test file. Error: {}".format(stderr))
+
+    def remove_test_script(self, name):
+        ret_code, stdout, stderr = self.run_command(
+            'gcloud compute ssh {} -- "rm {}"'.format(
+                name,
+                self.TEST_SCRIPT_FILE_NAME,
+            )
+        )
+        self.assertEqual(ret_code, 0, "Failed to remove test file. Error: {}".format(stderr))
+
+    @staticmethod
+    def run_command(command, timeout_in_minutes=DEFAULT_TIMEOUT):
+        print(command)
+        kill = lambda process: process.kill()
+        p = subprocess.Popen(
+            command,
+            shell=True,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        timeout = timeout_in_minutes * 60
+        my_timer = Timer(timeout, kill, [p])
+        try:
+            my_timer.start()
+            stdout, stderr = p.communicate()
+            stdout, stderr = stdout.decode('utf-8'), stderr.decode('utf-8')
+        finally:
+            my_timer.cancel()
+        logging.info("Ran %s: retcode: %d, stdout: %s, stderr: %s", command, p.returncode, stdout, stderr)
```

Comment:
Does this print a lot of spam when you're creating a cluster?

Reply:
Not really, but depends on `a lot` definition ;)
I believe this log is important - might held interesting info and by default INFO isn't logged - I added small line that allows to specify which log_level should be used: ac301e3226d14f457112afcdafbbf25bbb3f0452.
PTAL